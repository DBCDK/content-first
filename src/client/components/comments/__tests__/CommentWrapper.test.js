import React from 'react';
import {CommentWrapper} from '../CommentWrapper.component';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

jest.mock('react-textarea-autosize', () => 'textarea');

function generateComments({key = 'test', saving, error} = {}) {
  const comment = {
    comment: `comment_${key}`,
    _id: `${key}`,
    user: {
      name: 'Benny Cosmos',
      openplatformId: '1234'
    }
  };

  if (saving) {
    comment.saving = saving;
  }

  if (error) {
    comment.error = error;
  }

  return comment;
}

describe('CommentContainer', () => {
  test('Render comment', () => {
    const tree = renderer.create(
      <CommentWrapper comment={generateComments()} />
    );
    expect(tree).toMatchSnapshot();
  });
  test('Render edit form', () => {
    const comment = generateComments();
    const tree = renderer.create(
      <CommentWrapper comment={comment} user={comment.user} />
    );
    expect(tree).toMatchSnapshot();
  });
  test('Render save form', () => {
    const comment = generateComments({saving: true});
    const tree = renderer.create(
      <CommentWrapper comment={comment} user={comment.user} />
    );
    expect(tree).toMatchSnapshot();
  });
  test('Edit button', () => {
    const comment = generateComments();
    const tree = mount(
      <CommentWrapper comment={comment} user={comment.user} />
    );

    expect(tree).toMatchSnapshot();
    tree
      .find('.comment-edit-button')
      .hostNodes()
      .simulate('click');
    expect(tree).toMatchSnapshot();
  });
  test('Delete button', () => {
    const comment = generateComments();
    const deleteComment = jest.fn();
    const tree = mount(
      <CommentWrapper
        comment={comment}
        user={comment.user}
        deleteComment={deleteComment}
      />
    );
    expect(tree).toMatchSnapshot();
    tree
      .find('.comment-edit-button')
      .hostNodes()
      .simulate('click');
    tree
      .find('.comment-delete')
      .hostNodes()
      .simulate('click');
    expect(deleteComment.mock.calls[0][0]).toEqual(comment);
  });
  test('Cancel button', () => {
    const comment = generateComments();
    const tree = mount(
      <CommentWrapper comment={comment} user={comment.user} />
    );
    expect(tree).toMatchSnapshot();
    tree
      .find('.comment-edit-button')
      .hostNodes()
      .simulate('click');
    tree
      .find('.comment-cancel')
      .hostNodes()
      .simulate('click');
    expect(tree).toMatchSnapshot();
  });
  test('Save button', () => {
    const comment = generateComments();
    const editComment = jest.fn();
    const toggleComment = jest.fn();
    const tree = mount(
      <CommentWrapper
        comment={comment}
        user={comment.user}
        toggleEdit={toggleComment}
        editComment={editComment}
      />
    );
    expect(tree).toMatchSnapshot();
    tree
      .find('.comment-edit-button')
      .hostNodes()
      .simulate('click');
    tree
      .find('.comment-textarea')
      .simulate('change', {target: {value: 'some updated comment'}});
    tree
      .find('#comment-submit')
      .hostNodes()
      .simulate('click');
    expect(editComment.mock.calls[0][0]).toEqual({
      ...comment,
      comment: 'some updated comment'
    });
  });
});
