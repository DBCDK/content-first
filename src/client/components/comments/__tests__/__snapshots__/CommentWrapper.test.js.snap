// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer Cancel button 1`] = `
<CommentWrapper
  comment={
    Object {
      "_id": "test",
      "comment": "comment_test",
      "editing": true,
      "user": Object {
        "name": "Benny Cosmos",
        "openplatformId": "1234",
      },
    }
  }
  toggleEdit={[Function]}
  user={
    Object {
      "name": "Benny Cosmos",
      "openplatformId": "1234",
    }
  }
>
  <div
    className="comment-wrapper"
    key="test"
  >
    <button
      className="comment-edit-button btn btn-link link-subtle"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-pencil"
      />
    </button>
    <div
      className="flex mb2"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Component
        size="35"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        user={
          Object {
            "name": "Benny Cosmos",
            "openplatformId": "1234",
          }
        }
      >
        <div
          className="profile-card "
        >
          <span
            className="profile-image small round"
            style={
              Object {
                "height": "35px",
                "marginRight": "20px",
                "width": "35px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-user"
              style={
                Object {
                  "fontSize": "17.5px",
                }
              }
            />
          </span>
        </div>
      </Component>
      <div
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <div
          className="comment-author"
        >
          Benny Cosmos
        </div>
        <div
          className="comment-time mb1"
        >
          Lige nu
        </div>
        <CommentInput
          autoFocus={true}
          error={null}
          hideProfile={true}
          onCancel={[Function]}
          onChange={[Function]}
          onDelete={[Function]}
          onSubmit={[Function]}
          user={
            Object {
              "name": "Benny Cosmos",
              "openplatformId": "1234",
            }
          }
          value="comment_test"
        >
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="form-group "
              onFocus={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <textarea
                autoFocus={true}
                className="form-control mb1 comment-textarea"
                name="list-description"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Skriv kommentar"
                value="comment_test"
              />
              <div
                className="tr"
                style={
                  Object {
                    "height": "0px",
                    "opacity": 0,
                    "overflow": "hidden",
                    "transition": "opacity 200ms",
                  }
                }
              >
                <button
                  className="comment-delete btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Slet
                </button>
                <button
                  className="comment-cancel btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Fortryd
                </button>
                <button
                  className="btn btn-success"
                  disabled={false}
                  id="comment-submit"
                  onClick={[Function]}
                >
                  Gem
                </button>
              </div>
            </div>
          </div>
        </CommentInput>
      </div>
    </div>
  </div>
</CommentWrapper>
`;

exports[`CommentContainer Delete button 1`] = `
<CommentWrapper
  comment={
    Object {
      "_id": "test",
      "comment": "comment_test",
      "editing": true,
      "user": Object {
        "name": "Benny Cosmos",
        "openplatformId": "1234",
      },
    }
  }
  deleteComment={[Function]}
  user={
    Object {
      "name": "Benny Cosmos",
      "openplatformId": "1234",
    }
  }
>
  <div
    className="comment-wrapper"
    key="test"
  >
    <button
      className="comment-edit-button btn btn-link link-subtle"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-pencil"
      />
    </button>
    <div
      className="flex mb2"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Component
        size="35"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        user={
          Object {
            "name": "Benny Cosmos",
            "openplatformId": "1234",
          }
        }
      >
        <div
          className="profile-card "
        >
          <span
            className="profile-image small round"
            style={
              Object {
                "height": "35px",
                "marginRight": "20px",
                "width": "35px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-user"
              style={
                Object {
                  "fontSize": "17.5px",
                }
              }
            />
          </span>
        </div>
      </Component>
      <div
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <div
          className="comment-author"
        >
          Benny Cosmos
        </div>
        <div
          className="comment-time mb1"
        >
          Lige nu
        </div>
        <CommentInput
          autoFocus={true}
          error={null}
          hideProfile={true}
          onCancel={[Function]}
          onChange={[Function]}
          onDelete={[Function]}
          onSubmit={[Function]}
          user={
            Object {
              "name": "Benny Cosmos",
              "openplatformId": "1234",
            }
          }
          value="comment_test"
        >
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="form-group "
              onFocus={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <textarea
                autoFocus={true}
                className="form-control mb1 comment-textarea"
                name="list-description"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Skriv kommentar"
                value="comment_test"
              />
              <div
                className="tr"
                style={
                  Object {
                    "height": "0px",
                    "opacity": 0,
                    "overflow": "hidden",
                    "transition": "opacity 200ms",
                  }
                }
              >
                <button
                  className="comment-delete btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Slet
                </button>
                <button
                  className="comment-cancel btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Fortryd
                </button>
                <button
                  className="btn btn-success"
                  disabled={false}
                  id="comment-submit"
                  onClick={[Function]}
                >
                  Gem
                </button>
              </div>
            </div>
          </div>
        </CommentInput>
      </div>
    </div>
  </div>
</CommentWrapper>
`;

exports[`CommentContainer Edit button 1`] = `
<CommentWrapper
  comment={
    Object {
      "_id": "test",
      "comment": "comment_test",
      "user": Object {
        "name": "Benny Cosmos",
        "openplatformId": "1234",
      },
    }
  }
  toggleEdit={[Function]}
  user={
    Object {
      "name": "Benny Cosmos",
      "openplatformId": "1234",
    }
  }
>
  <div
    className="comment-wrapper"
    key="test"
  >
    <button
      className="comment-edit-button btn btn-link link-subtle"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-pencil"
      />
    </button>
    <div
      className="flex mb2"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Component
        size="35"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        user={
          Object {
            "name": "Benny Cosmos",
            "openplatformId": "1234",
          }
        }
      >
        <div
          className="profile-card "
        >
          <span
            className="profile-image small round"
            style={
              Object {
                "height": "35px",
                "marginRight": "20px",
                "width": "35px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-user"
              style={
                Object {
                  "fontSize": "17.5px",
                }
              }
            />
          </span>
        </div>
      </Component>
      <div
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <div
          className="comment-author"
        >
          Benny Cosmos
        </div>
        <div
          className="comment-time mb1"
        >
          Lige nu
        </div>
        <div
          className="comment"
          dangerouslySetInnerHTML={
            Object {
              "__html": "comment_test",
            }
          }
        />
      </div>
    </div>
  </div>
</CommentWrapper>
`;

exports[`CommentContainer Render comment 1`] = `
<div
  className="comment-wrapper"
>
  
  <div
    className="flex mb2"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="profile-card "
    >
      <span
        className="profile-image small round"
        style={
          Object {
            "height": "35px",
            "marginRight": "20px",
            "width": "35px",
          }
        }
      >
        <span
          className="glyphicon glyphicon-user"
          style={
            Object {
              "fontSize": "17.5px",
            }
          }
        />
      </span>
      
    </div>
    <div
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <div
        className="comment-author"
      >
        Benny Cosmos
      </div>
      <div
        className="comment-time mb1"
      >
        Lige nu
      </div>
      <div
        className="comment"
        dangerouslySetInnerHTML={
          Object {
            "__html": "comment_test",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`CommentContainer Render edit form 1`] = `
<div
  className="comment-wrapper"
>
  
  <button
    className="comment-edit-button btn btn-link link-subtle"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-pencil"
    />
  </button>
  <div
    className="flex mb2"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="profile-card "
    >
      <span
        className="profile-image small round"
        style={
          Object {
            "height": "35px",
            "marginRight": "20px",
            "width": "35px",
          }
        }
      >
        <span
          className="glyphicon glyphicon-user"
          style={
            Object {
              "fontSize": "17.5px",
            }
          }
        />
      </span>
      
    </div>
    <div
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <div
        className="comment-author"
      >
        Benny Cosmos
      </div>
      <div
        className="comment-time mb1"
      >
        Lige nu
      </div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="form-group "
          onFocus={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <textarea
            autoFocus={true}
            className="form-control mb1 comment-textarea"
            disabled={undefined}
            name="list-description"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Skriv kommentar"
            value="comment_test"
          />
          
          <div
            className="tr"
            style={
              Object {
                "height": "0px",
                "opacity": 0,
                "overflow": "hidden",
                "transition": "opacity 200ms",
              }
            }
          >
            <button
              className="comment-delete btn btn-link link-subtle"
              onClick={[Function]}
            >
              Slet
            </button>
            <button
              className="comment-cancel btn btn-link link-subtle"
              onClick={[Function]}
            >
              Fortryd
            </button>
            <button
              className="btn btn-success"
              disabled={false}
              id="comment-submit"
              onClick={[Function]}
            >
              Gem
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CommentContainer Render save form 1`] = `
<div
  className="comment-wrapper"
>
  <div
    className="comment-saving"
  >
    <span
      className="spinner"
      style={
        Object {
          "borderTopColor": null,
          "display": "inline-block",
          "height": "30px",
          "width": "30px",
        }
      }
    />
  </div>
  <button
    className="comment-edit-button btn btn-link link-subtle"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-pencil"
    />
  </button>
  <div
    className="flex mb2"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="profile-card "
    >
      <span
        className="profile-image small round"
        style={
          Object {
            "height": "35px",
            "marginRight": "20px",
            "width": "35px",
          }
        }
      >
        <span
          className="glyphicon glyphicon-user"
          style={
            Object {
              "fontSize": "17.5px",
            }
          }
        />
      </span>
      
    </div>
    <div
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <div
        className="comment-author"
      >
        Benny Cosmos
      </div>
      <div
        className="comment-time mb1"
      >
        Lige nu
      </div>
      <div
        className="comment"
        dangerouslySetInnerHTML={
          Object {
            "__html": "comment_test",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`CommentContainer Save button 1`] = `
<CommentWrapper
  comment={
    Object {
      "_id": "test",
      "comment": "comment_test",
      "editing": true,
      "user": Object {
        "name": "Benny Cosmos",
        "openplatformId": "1234",
      },
    }
  }
  editComment={[Function]}
  toggleEdit={[Function]}
  user={
    Object {
      "name": "Benny Cosmos",
      "openplatformId": "1234",
    }
  }
>
  <div
    className="comment-wrapper"
    key="test"
  >
    <button
      className="comment-edit-button btn btn-link link-subtle"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-pencil"
      />
    </button>
    <div
      className="flex mb2"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Component
        size="35"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        user={
          Object {
            "name": "Benny Cosmos",
            "openplatformId": "1234",
          }
        }
      >
        <div
          className="profile-card "
        >
          <span
            className="profile-image small round"
            style={
              Object {
                "height": "35px",
                "marginRight": "20px",
                "width": "35px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-user"
              style={
                Object {
                  "fontSize": "17.5px",
                }
              }
            />
          </span>
        </div>
      </Component>
      <div
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <div
          className="comment-author"
        >
          Benny Cosmos
        </div>
        <div
          className="comment-time mb1"
        >
          Lige nu
        </div>
        <CommentInput
          autoFocus={true}
          error={null}
          hideProfile={true}
          onCancel={[Function]}
          onChange={[Function]}
          onDelete={[Function]}
          onSubmit={[Function]}
          user={
            Object {
              "name": "Benny Cosmos",
              "openplatformId": "1234",
            }
          }
          value="comment_test"
        >
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="form-group "
              onFocus={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <textarea
                autoFocus={true}
                className="form-control mb1 comment-textarea"
                name="list-description"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Skriv kommentar"
                value="comment_test"
              />
              <div
                className="tr"
                style={
                  Object {
                    "height": "0px",
                    "opacity": 0,
                    "overflow": "hidden",
                    "transition": "opacity 200ms",
                  }
                }
              >
                <button
                  className="comment-delete btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Slet
                </button>
                <button
                  className="comment-cancel btn btn-link link-subtle"
                  onClick={[Function]}
                >
                  Fortryd
                </button>
                <button
                  className="btn btn-success"
                  disabled={false}
                  id="comment-submit"
                  onClick={[Function]}
                >
                  Gem
                </button>
              </div>
            </div>
          </div>
        </CommentInput>
      </div>
    </div>
  </div>
</CommentWrapper>
`;
